numerics/fixed_arrays.hpp

```diff
@@ -41,8 +41,11 @@ class FixedVector final {
  private:
   std::array<Scalar, size_> data_;
 
+  template<typename L, typename R, int s>
+  friend constexpr Product<L, R> operator*(FixedVector<L, s> const& left,
+                                           FixedVector<R, s> const& right);
```

Comment:
Implicit transposition tends to be very messy; in particular we lose associativity: A(vv) is a matrix, (Av)v is a scalar.

We could have a DotProduct, but that is going to make the code look less like the references, which is bad from the point of view of formula translation. 
Let us go with
```C++
template<typename V>
struct TransposedView {
  V const& transpose;
};
```
In `FixedVector`:
```C++
TransposedView<FixedVector> Transpose() const {
  return {.transpose = *this};
}
```
And then this operator becomes:
```C++
  template<typename L, typename R, int s>
  friend constexpr Product<L, R> operator*(TransposedView<FixedVector<L, s>> const& left,
                                           FixedVector<R, s> const& right);
```

Reply:
(Note: we may want to have a more general and more powerful `TransposedView` in the future as the result of all `Transpose` (with conversions to the appropriate matrix types if actual transposition of the data is needed), but this is a rabbit hole for another day; for now this draft is good enough.)